src/Expr.cpp

```diff
@@ -0,0 +1,12 @@
+#include "Expr.h"
+#include "IR.h"
+
+namespace Halide {
+// FIXME: This feels like a horrible hack. FloatImm should be able
```

Comment:
See Simplify.cpp line 301. That's how the constant folding would look with
the cast solution - you'd dig out the underlying floats with
const_castfloat, combine them as floats in whatever way is correct for
float16s, then rewrap in a cast to float16.



On Wed, Aug 19, 2015 at 10:57 AM, Dan Liew <notifications@github.com> wrote:

> In src/Expr.cpp
> <https://github.com/halide/Halide/pull/904#discussion_r37445987>:
>
> > @@ -0,0 +1,12 @@
> > +#include "Expr.h"
> > +#include "IR.h"
> > +
> > +namespace Halide {
> > +// FIXME: This feels like a horrible hack. FloatImm should be able
>
> @abadams <https://github.com/abadams> Although "Use a cast of a float" is
> the least invasive change I'm leaning (and I'm going to try) Making
> FloatImm more general to serve a rare use-case, affecting every use of it
> because I need to know where it is going to be used (e.g. constant folding
> needs to be done differently) at the same time I could also add support for
> representing double immediates.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/halide/Halide/pull/904/files#r37445987>.
>